backend/logic/printful/sync/stockUpdate.js

```diff
@@ -0,0 +1,66 @@
+const _get = require('lodash/get')
+const program = require('commander')
+const { decryptConfig } = require('../../../utils/encryptedConfig')
+
+const { get } = require('../api')
+const { Product } = require('../../../models')
+
+/**
+ * Updates discontinued variants and variants that are out of stock to a collection.
+ * And creates collection if it doesn't exist
+ *
+ * @param {Object} shop
+ * @param {Number|String} productId
+ * @param {Array<Object>} variants
+ */
+const stockUpdate = async (shop, productId, variants) => {
+  const shopConfig = decryptConfig(shop.config)
+  const shopInventory = _get(shopConfig, 'inventory')
+  if (shopInventory) {
+    const variantsStock = variants.reduce(
+      (stockData, variant) => ({
+        ...stockData,
+        [variant]: 0
+      }),
+      {}
+    )
+
+    const apiKey = shopConfig.printful
+    let existingVariants = 0
+
+    if (apiKey) {
+      const productData = await get(`/store/products/${productId}`, {
+        apiKey: apiKey
+      })
+      existingVariants = _get(productData, 'result.sync_product.synced', 0) - variants.length
+      existingVariants = existingVariants < 0 ? 0 : existingVariants
+    }
+
+    const updatedStockData = {
+      productId: productId.toString(),
+      shopId: shop.id,
+      stockLeft: existingVariants * 10000,
+      variantsStock
+    }
+
+    let existingDbItem
+
+    if (productId) {
+      existingDbItem = await Product.findOne({
+        where: {
+          productId: productId.toString(),
+          shopId: shop.id
+        }
+      })
+    }
+
+    if (existingDbItem) {
+      await existingDbItem.update(updatedStockData)
+    } else {
+      await Product.create(updatedStockData)
```

Comment:
Sorry, I'm not very familiar with that area of our codebase, but I still don't understand exactly why we need to create a Product row here. Under which scenario would we need this? I would expect that for Printful to call us back on this webhook, a product row must have already been created already, earlier on when the product was added to the store in the first place?
If that holds, then I'd consider receiving a webhook call for a product row that does not exit to be an error that we log but we don't create a product database row.

Also note that adding Product rows in the database is not enough to add products to the store. For that to happen, static json files also need to be stored on disk under the store's data directory. See code from the upserProduct method [here](https://github.com/OriginProtocol/dshop/blob/master/backend/logic/products/index.js#L353).

Reply:
> I would expect that for Printful to call us back on this webhook, a product row must have already been created already, earlier on when the product was added to the store in the first place?

The DB items are not created in the sync
https://github.com/OriginProtocol/dshop/blob/cc6782b0c0e989feee9bc35bbf5d2c7ee37a7db4/backend/logic/printful/sync/writeProductData.js#L52

> Also note that adding Product rows in the database is not enough to add products to the store. 

I am not adding the new product here, the table contains only stock data

Reply:
Cleared this up on discord..